src/r1cs/constraint_system.rs

```diff
@@ -104,9 +104,12 @@ pub trait RandomizableConstraintSystem: ConstraintSystem {
     ///     // ...
     /// })
     /// ```
-    fn specify_randomized_constraints<F>(&mut self, callback: F) -> Result<(), R1CSError>
+    fn specify_randomized_constraints<'a: 'constraints, F>(
+        &mut self,
```

Comment:
Not specifying an explicit lifetime on `&mut self` might get people into trouble [with late-bound lifetimes](https://github.com/rust-lang/rust/issues/42868). I've had that while experimenting on it, but haven't come to fix it yet. I think it should be possible to add it later though, without breaking API...

Reply:
Dismissed to #323.